ona/tests.py

```diff
@@ -0,0 +1,168 @@
+import datetime
+import responses
+from django.contrib.auth.models import User
+from django.test import TestCase
+from django.db.models.signals import post_save
+
+from reports.models import Report, Category, ProjectCategory, Location
+from accounts.models import Project, UserProject, Integration
+from .models import Submission, fire_subm_action_if_undelivered
+
+
+class SubmissionsTestCase(TestCase):
+    def _replace_post_save_hooks(self):
+        has_listeners = lambda: post_save.has_listeners(Submission)
+        assert has_listeners(), (
+            "Submission model has no post_save listeners. Make sure helpers"
+            " cleaned up properly in earlier tests.")
+        post_save.disconnect(fire_subm_action_if_undelivered,
+                             sender=Submission)
+        assert not has_listeners(), (
+            "Submission model still has post_save listeners. Make sure helpers"
+            " cleaned up properly in earlier tests.")
+
+    def _restore_post_save_hooks(self):
+        has_listeners = lambda: post_save.has_listeners(Submission)
+        assert not has_listeners(), (
+            "Submission model still has post_save listeners. Make sure helpers"
+            " removed them properly in earlier tests.")
+        post_save.connect(fire_subm_action_if_undelivered, sender=Submission)
+
+    def setUp(self):
+        super(SubmissionsTestCase, self).setUp()
+        self._replace_post_save_hooks()
+        self.normalusername = 'testnormaluser'
+        self.normalpassword = 'testnormalpass'
+        self.normaluser = User.objects.create_user(
+            self.normalusername,
+            'testnormaluser@example.com',
+            self.normalpassword)
+        self.project = self.make_user_project()
+
+    def tearDown(self):
+        self._restore_post_save_hooks()
+
+
+class TestSubmissions(SubmissionsTestCase):
+    def make_ona_integration(self, project, activate=True):
+        ona_integration = Integration.objects.create(
+            project=project, integration_type="Ona", details={
+                "url": "https://ona.io/api/v1/submissions",
+                "username": "testuser",
+                "password": "testuserpass",
+                "form_id": "test_form"
+            },
+            active=activate
+        )
+        ona_integration.save()
+        return ona_integration
+
+    def make_user_project(self):
+        project3 = Project.objects.create(
+            code="TESTPROJ3", name="Test Project 3")
+        userproject = UserProject.objects.create(
+            user=self.normaluser)
+        userproject.projects.add(project3)
+        userproject.save()
+        return project3
```

Comment:
I think we can call the variable `project3` just `project` since it's the only `project` in this method?